brokers/unified/broker.go

```diff
@@ -125,10 +141,41 @@ func (b *Broker) ProcessPlugins(metas []model.PluginMeta) error {
 	return nil
 }
 
+// getPluginMetas downloads the metadata for each plugin in plugins. If specified,
+// defaultRegistry is used as the registry for plugins that do not specify their registry.
+// If defaultRegistry is empty, and any plugin does not specify a registry, an error is returned.
+func (b *Broker) getPluginMetas(plugins []model.PluginFQN, defaultRegistry string) ([]model.PluginMeta, error) {
+	metas := make([]model.PluginMeta, len(plugins))
+	for idx, plugin := range plugins {
+		log.Printf("Fetching plugin meta.yaml for %s:%s", plugin.ID, plugin.Version)
+		var registry string
+		if plugin.Registry != "" {
+			registry = strings.TrimSuffix(plugin.Registry, "/")
+		} else {
+			if defaultRegistry == "" {
+				return nil, fmt.Errorf("plugin '%s' does not specify registry and no default is provided", plugin.ID)
+			}
+			registry = strings.TrimSuffix(defaultRegistry, "/")
+		}
+		pluginURL := fmt.Sprintf("%s/%s/%s/meta.yaml", registry, plugin.ID, plugin.Version)
```

Comment:
Taking into account that Che properties specify plugin registry without `/plugins` suffix (`che.workspace.plugin_registry_url=https://che-plugin-registry.openshift.io`) it would be better to align them and add `/plugins` here too. Treat it as API, not just file hosting.

Reply:
Since in a case of old plugin `/plugins` can be treated as `publisher` we probbaly can not avoid setting plugin registry without `/plugins`

Reply:
I've pulled the URL format into a const at the top, but now I'm thinking whether it makes sense to include it in the config at some point. There would also be the problem of using multiple registries for different plugins; they would all have to conform to our current spec which could be tricky.